instrumentation/activej-http-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/activejhttp/ActivejHttpServerConnectionInstrumentation.java

```diff
@@ -0,0 +1,114 @@
+  }
+
+  @SuppressWarnings("unused")
+  public static class ServeAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void methodEnter(
+        @Advice.This AsyncServlet asyncServlet,
+        @Advice.Argument(0) HttpRequest request,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope,
+        @Advice.Local("httpRequest") HttpRequest httpRequest) {
+      Context parentContext = currentContext();
+      httpRequest = request;
+      if (!instrumenter().shouldStart(parentContext, request)) {
+        return;
+      }
+      context = instrumenter().start(parentContext, request);
+      scope = context.makeCurrent();
+    }
+
+    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
+    public static void methodExit(
+        @Advice.This AsyncServlet asyncServlet,
+        @Advice.Return(readOnly = false) Promise<HttpResponse> responsePromise,
+        @Advice.Thrown Throwable throwable,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope,
+        @Advice.Local("httpRequest") HttpRequest httpRequest) {
+      if (context == null || scope == null || httpRequest == null) {
+        return;
+      }
+      String traceId = Span.fromContext(context).getSpanContext().getTraceId();
+      String spanId = Span.fromContext(context).getSpanContext().getSpanId();
+      String traceFlags =
+          Span.fromContext(context).getSpanContext().getTraceFlags().asHex().substring(0, 2);
+      String traceparent = String.format("00-%s-%s-%s", traceId, spanId, traceFlags);
+
+      scope.close();
+      String traceparentHeader = "traceparent";
+      if (responsePromise != null) {
+        HttpResponse httpResponse = responsePromise.getResult();
+        Throwable error = throwable;
+        if (responsePromise.isException()) {
+          error = responsePromise.getException();
+        }
+        httpResponse = ActivejHttpServerHelper.createResponse(error, traceparent, httpResponse);
+        instrumenter().end(context, httpRequest, httpResponse, error);
+        responsePromise = Promise.of(httpResponse);
+      } else if (throwable != null) {
+        HttpResponse httpResponse =
+            HttpResponse.builder()
+                .withCode(500)
+                .withPlainText(throwable.getMessage())
+                .withHeader(HttpHeaders.of(traceparentHeader), traceparent)
+                .build();
+        instrumenter().end(context, httpRequest, httpResponse, throwable);
+        responsePromise = Promise.of(httpResponse);
+      } else {
+        HttpResponse httpResponse =
+            HttpResponse.notFound404()
+                .withHeader(HttpHeaders.of(traceparentHeader), traceparent)
+                .build();
+        instrumenter().end(context, httpRequest, httpResponse, throwable);
+        responsePromise = Promise.of(httpResponse);
+      }
+    }
```

Comment:
This doesn't make sense to me. I'd suggest you rewrite it similar to
```java
if (scope != null) {
  return;
}
scope.close();
if (throwable != null) {
  instrumenter().end(context, httpRequest, null, throwable);
} else {
  responsePromise = ActivejHttpServerUtil.wrap(responsePromise, context, httpRequest);
}
```
you shouldn't use fancy stuff like lambdas or inner classes in an advice class
```java
  public static Promise<HttpResponse> wrap(Promise<HttpResponse> promise, Context context, HttpRequest httpRequest) {
    return promise.whenComplete((httpResponse, exception) -> {
      instrumenter().end(context, httpRequest, httpResponse, exception);
    });
  }
```